ports/servoshell/egl/ohos.rs

```diff
@@ -386,32 +387,38 @@ extern "C" fn on_dispatch_key_event(xc: *mut OH_NativeXComponent, _window: *mut
     }
 }
 
+static LOGGER: LazyLock<hilog::Logger> = LazyLock::new(|| {
+    let mut builder = hilog::Builder::new();
+    let filters = [
+        "fonts",
+        "servo",
+        "servoshell",
+        "servoshell::egl:gl_glue",
+        // Show redirected stdout / stderr by default
+        "servoshell::egl::log",
+        // Show JS errors by default.
+        "script::dom::bindings::error",
+        // Show GL errors by default.
+        "canvas::webgl_thread",
+        "compositing::compositor",
+        "compositing::touch",
+        "constellation::constellation",
+        "ohos_ime",
+    ];
+    for &module in &filters {
+        builder.filter_module(module, log::LevelFilter::Debug);
+    }
+
+    builder.filter_level(LevelFilter::Warn).build()
+});
+
 fn initialize_logging_once() {
     static ONCE: Once = Once::new();
     ONCE.call_once(|| {
-        let mut builder = hilog::Builder::new();
-        let filters = [
-            "fonts",
-            "servo",
-            "servoshell",
-            "servoshell::egl:gl_glue",
-            // Show redirected stdout / stderr by default
-            "servoshell::egl::log",
-            // Show JS errors by default.
-            "script::dom::bindings::error",
-            // Show GL errors by default.
-            "canvas::webgl_thread",
-            "compositing::compositor",
-            "compositing::touch",
-            "constellation::constellation",
-            "ohos_ime",
-        ];
-        for &module in &filters {
-            builder.filter_module(module, log::LevelFilter::Debug);
-        }
-
-        builder.filter_level(LevelFilter::Warn).init();
-
+        let logger: &'static hilog::Logger = &LOGGER;
+        let max_level = logger.filter();
+        let r = log::set_logger(logger).and_then(|()| Ok(log::set_max_level(max_level)));
+        debug!("Attempted to register the logger: {r:?} and set max level to: {max_level}");
         info!("Servo Register callback called!");
```

Comment:
This segment has been written manually instead of relying on `hilog::Builder::{init, try_init}` methods as the logger will already be initialized at this point.